src/main/java/com/adobe/cq/commerce/graphql/client/impl/GraphqlClientImpl.java

```diff
@@ -359,24 +374,16 @@ private HttpUriRequest buildRequest(GraphqlRequest request, RequestOptions optio
             rb.setEntity(new StringEntity(gson.toJson(request), StandardCharsets.UTF_8.name()));
         }
 
-        if (configuration.httpHeaders() != null) {
-            for (String httpHeader : configuration.httpHeaders()) {
-                // We ignore empty values, this may happen because of the way the AEM OSGi configuration editor works
-                if (StringUtils.isBlank(httpHeader)) {
-                    continue;
-                }
-
-                int idx = httpHeader.indexOf(":");
-                if (idx < 1 || httpHeader.length() <= (idx + 1)) {
-                    throw new IllegalStateException("The HTTP header is not a name:value pair --> " + httpHeader);
-                }
-                rb.addHeader(httpHeader.substring(0, idx).trim(), httpHeader.substring(idx + 1).trim());
+        for (String httpHeader : configuration.httpHeaders()) {
+            String[] parts = StringUtils.split(httpHeader, ":", 2);
+            if (parts.length == 2 && StringUtils.isNoneBlank(parts[0], parts[1])) {
+                rb.addHeader(parts[0].trim(), parts[1].trim());
             }
```

Comment:
To make debugging easier (since you're removing the exception) it would be useful to detect early, probably in GraphqlClientConfigurationImpl, that there are invalid header configurations and log a warning for it.

Reply:
This is done here 

https://github.com/adobe/commerce-cif-graphql-client/pull/33/files#diff-65f3d41f0df4ce50c46d0519093fab1f3edb026a692c380e346c81d7dd1203f1R160-R162